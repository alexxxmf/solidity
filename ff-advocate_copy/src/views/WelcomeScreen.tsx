import React, { Component } from "react";
import {
  Image,
  ImageBackground,
  StyleSheet,
  Text,
  View,
  StatusBar,
  Platform
} from "react-native";
import Button from "./components/Button";
import ApplicationStyles from "./styles/ApplicationStyles";
import { Metrics, Images } from "../themes";
import { AppScreenProps } from "../types";
import { NavigationInjectedProps, withNavigation } from "react-navigation";

type Props = {
  screenProps: AppScreenProps;
} & NavigationInjectedProps<{}>;

class WelcomeScreen extends Component<Props> {
  render() {
    const { userName } = this.props.screenProps;

    return (
      <View style={ApplicationStyles.mainContainer}>
        {Platform.OS === "ios" ? <StatusBar barStyle="dark-content" /> : null}
        <View
          style={[ApplicationStyles.sectionContainer, styles.welcomeSection]}
        >
          <Text style={[ApplicationStyles.sectionSubTitle, styles.sectionText]}>
            Hello {userName}
          </Text>
          <Text style={[ApplicationStyles.sectionSubTitle, styles.sectionText]}>
            What do you want to do today?
          </Text>
        </View>
        <ImageBackground
          source={Images.idea}
          imageStyle={{ resizeMode: "cover" }}
          style={styles.callToAction}
        >
          <View>
            <Image
              source={Images.prepare}
              style={{
                height: Metrics.images.xxlarge,
                width: Metrics.images.xxlarge
              }}
              resizeMode="contain"
            />
            <Button
              title="Prepare"
              type="outline"
              onPress={() => {
                this.props.navigation.navigate("PrepareSessions");
              }}
            />
          </View>
          <View>
            <Image
              source={Images.deliver}
              style={{
                height: Metrics.images.xxlarge,
                width: Metrics.images.xxlarge
              }}
              resizeMode="contain"
            />
            <Button
              title="Deliver"
              type="outline"
              onPress={() => {
                this.props.navigation.navigate("DeliverSessions");
              }}
            />
          </View>
        </ImageBackground>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  welcomeSection: {
    paddingTop: Metrics.doubleSection
  },
  sectionText: {
    textAlign: "center"
  },
  callToAction: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    paddingHorizontal: Metrics.section,
    paddingBottom: Metrics.screenHeight / 4,
    flex: 1
  }
});

export default withNavigation<Props>(WelcomeScreen);
