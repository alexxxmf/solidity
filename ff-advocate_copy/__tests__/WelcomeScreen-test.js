/**
 * @jest-environment jsdom
 */
import React from "react";
import WelcomeScreen from "../src/views/WelcomeScreen";

import renderer from "react-test-renderer";

jest.mock("rollbar-react-native", () => {
  return {
    Client: function Client(data) {
      return { data, logError: console.warn("mocked") };
    }
  };
});

test("renders correctly", () => {
  const f = jest.fn();
  const tree = renderer
    .create(<WelcomeScreen navigation={f} screenProps={{ userName: null }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
