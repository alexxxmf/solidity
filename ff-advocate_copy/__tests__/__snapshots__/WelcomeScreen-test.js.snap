// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F0F7FF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 10,
        },
        Object {
          "paddingTop": 50,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Raleway",
            "fontSize": 20,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      Hello 
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Raleway",
            "fontSize": 20,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      What do you want to do today?
    </Text>
  </View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 333.5,
        "paddingHorizontal": 25,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/welcome/idea.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {
            "resizeMode": "cover",
          },
        ]
      }
    />
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/images/welcome/prepare.png",
          }
        }
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      />
      <View
        style={
          Object {
            "borderRadius": 8,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#111111",
                "borderRadius": 8,
                "borderWidth": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#111111",
                  "fontFamily": "Raleway",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Prepare
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View>
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/images/welcome/deliver.png",
          }
        }
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      />
      <View
        style={
          Object {
            "borderRadius": 8,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#111111",
                "borderRadius": 8,
                "borderWidth": 2,
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 8,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#111111",
                  "fontFamily": "Raleway",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "paddingBottom": 1,
                  "paddingTop": 2,
                  "textAlign": "center",
                }
              }
            >
              Deliver
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
